Print more information for BlockEncodingBuffer.toString()

yingsu00 opened the pull request:
Printing more information for BlockEncodingBuffer.toString() makes the debugging easier.

```
== NO RELEASE NOTE ==
```

mbasmanova commented on presto-main/src/main/java/com/facebook/presto/operator/repartition/LongArrayBlockEncodingBuffer.java:
```diff
@@ -118,6 +118,7 @@ public long getSerializedSizeInBytes()
     public String toString()
     {
         return toStringHelper(this)
+                .add("super", super.toString())
```
I don't think you need to call toString() explicitly here (and in other similar places).

yingsu00 replied:
@mbasmanova Actually without the .toString() the compilation would fail. The `add` method takes Object not String specifically:
`public ToStringHelper add(String name, @Nullable Object value)`

mbasmanova replied:
@yingsu00 I'm confused. Are you saying `.add("super", super)` fails to compile? What's the error? I expect this to work and call "super.toString()" internally.

yingsu00 replied:
> @yingsu00 I'm confused. Are you saying `.add("super", super)` fails to compile? What's the error? I expect this to work and call "super.toString()" internally.

Yes. Changing the code to 
```
@Override
    public String toString()
    {
        return toStringHelper(this)
                .add("super", super)
                .add("estimatedValueBufferMaxCapacity", estimatedValueBufferMaxCapacity)
                .add("valuesBufferCapacity", valuesBuffer == null ? 0 : valuesBuffer.length)
                .add("valuesBufferIndex", valuesBufferIndex)
                .toString();
    }
```
would get the following error message:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project presto-main: Compilation failure: Compilation failure:
[ERROR] /Users/yingsu/presto5/presto/presto-main/src/main/java/com/facebook/presto/operator/repartition/ByteArrayBlockEncodingBuffer.java:[142,36] '.' expected
[ERROR] /Users/yingsu/presto5/presto/presto-main/src/main/java/com/facebook/presto/operator/repartition/ByteArrayBlockEncodingBuffer.java:[146,28] ')' expected
```
My understanding is `super` cannot be used by itself. It has to be used in a constructor like `super()` or `super.method()`.

However I removed other .toString() calls on decodedBlock, valueBuffers, etc.

yingsu00 commented:
@mbasmanova Hi Masha, I corrected the typo, and also changed the last commit message from `Print estimatedNullsBufferMaxCapacity in BlockEncodingBuffers.toString()`to 
`Print estimated buffer max capacities in BlockEncodingBuffers.toString()` since it also prints the max capacity for other buffers. I will ping you once the tests turn green. Thank you for reviewing!
